pride-core/src/main/java/com/prezi/pride/vcs/RepoCache.java

```diff
@@ -42,7 +45,17 @@ public void checkoutThroughCache(VcsSupport vcsSupport, final String repositoryU
 			vcsSupport.checkout(repositoryUrl, moduleInCache, null, false, true);
 		} else {
 			log.info("Updating cached repository in " + moduleInCacheName);
-			vcsSupport.update(moduleInCache, null, false, true);
+			try {
+				vcsSupport.update(moduleInCache, null, false, true);
+			} catch (PrideException e) {
+				/*
+				 * Probably happended because mercurial repos (origin vs. repo in cache)
+				 * are unrelated. Remedy by reinitializing the cache.
+				 */
+				log.info("Cleaning up stale repo ({}) from the cache: {}", repositoryUrl, moduleInCacheName);
+				FileUtils.deleteDirectory(moduleInCache);
+				vcsSupport.checkout(repositoryUrl, moduleInCache, null, false, true);
+			}
```

Comment:
I'd like to avoid relying on exceptions here. What is the exact condition you are handling here? How can we work around it, or handle it more gracefully than indiscriminately catching all `PrideException`s?